retail/interactive-tutorial/RetailProducts.Samples/SetInventorySample.cs

```diff
@@ -0,0 +1,162 @@
+    /// <returns>Retail product with updated inventory info.</returns>
+    private static Product GetProductWithInventoryInfo(string productName)
+    {
+        var priceInfo = new PriceInfo
+        {
+            Price = 15.0f,
+            OriginalPrice = 20.0f,
+            Cost = 8.0f,
+            CurrencyCode = "USD"
+        };
+
+        var fulfillmentInfo = new FulfillmentInfo
+        {
+            Type = "pickup-in-store"
+        };
+
+        string[] placeIds = { "store1", "store2" };
+
+        fulfillmentInfo.PlaceIds.Add(placeIds);
+
+        var product = new Product
+        {
+            Name = productName,
+            PriceInfo = priceInfo,
+            Availability = Product.Types.Availability.InStock
+        };
+
+        product.FulfillmentInfo.Add(fulfillmentInfo);
+
+        return product;
+    }
+
+    /// <summary>
+    /// Get the set inventory request.
+    /// </summary>
+    /// <param name="productName">The actual product with inventory info.</param>
+    /// <returns>Set inventory request.</returns>
+    private static SetInventoryRequest GetSetInventoryRequest(string productName)
+    {
+        // The request timestamp
+        DateTime requestTimeStamp = DateTime.Now.ToUniversalTime();
+
+        // The out-of-order request timestamp
+        // request_time = datetime.datetime.now() - datetime.timedelta(days=1)
+
+        string[] paths = { "price_info", "availability", "fulfillment_info", "available_quantity" };
+        var setMask = new FieldMask();
+        setMask.Paths.AddRange(paths);
+
+        var setInventoryRequest = new SetInventoryRequest
+        {
+            Inventory = GetProductWithInventoryInfo(productName),
+            SetTime = Timestamp.FromDateTime(requestTimeStamp),
+            AllowMissing = true,
+            SetMask = setMask
+        };
+
+        var jsonSerializeSettings = new JsonSerializerSettings
+        {
+            NullValueHandling = NullValueHandling.Ignore,
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+            Formatting = Formatting.Indented
+        };
+
+        var setInventoryRequestJson = JsonConvert.SerializeObject(setInventoryRequest, jsonSerializeSettings);
+
+        Console.WriteLine("\nSet inventory request: \n\n" + setInventoryRequestJson);
+        return setInventoryRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to set product inventory.
+    /// </summary>
+    /// <param name="product">The actual product.</param>
+    public static void SetProductInventory(string productName)
+    {
+        var setInventoryRequest = GetSetInventoryRequest(productName);
+        GetProductServiceClient().SetInventory(setInventoryRequest);
+
+        // This is a long running operation and its result is not immediately present with get operations,
+        // thus we simulate wait with sleep method.
+        Console.WriteLine("\nSet inventory. Wait 50 seconds:\n");
+        Thread.Sleep(50000);
```

Comment:
Please don't this as there's no need, the library offers mechanism for polling long running operations:

```suggestion
        var setInventoryRequest = GetSetInventoryRequest(productName);
        var operation = GetProductServiceClient().SetInventory(setInventoryRequest);

        operation = operation.PollUntilCompleted();
```

Reply:
done